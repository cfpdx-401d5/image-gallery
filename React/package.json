{
  "name": "image-gallery",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "body-parser": "^1.17.1",
    "cors": "^2.8.1",
    "express": "^4.15.2",
    "mongoose": "^4.8.7",
    "morgan": "^1.8.1",
    "react": "^15.4.2",
    "react-dom": "^15.4.2"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-http": "^3.0.0",
    "enzyme": "^2.7.1",
    "enzyme-to-json": "^1.5.0",
    "eslint": "^3.17.1",
    "eslint-plugin-react": "^6.10.0",
    "mocha": "^3.2.0",
    "nodemon": "^1.11.0",
    "react-addons-test-utils": "^15.4.2",
    "react-scripts": "0.9.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "start-server": "nodemon server.js",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
