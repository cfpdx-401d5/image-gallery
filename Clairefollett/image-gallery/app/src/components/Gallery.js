import React from 'react';

export default function Gallery(props) {
    const galleryObj = props.puppies.map(puppies => {
        return (
            <GalleryDisplay key={puppies._id} puppies={puppies} onDelete={props.onDelete} />
        );
    });
    return (
        <ul>{galleryObj}</ul>
    );
}

function GalleryDisplay(props) {  
        return (
            <div className="clearfix">
                <li>
                    <p>Title: {props.puppies.title}</p>
                    <img className='gallery-img' src={props.puppies.url} alt='' />
                    <p>Description: {props.puppies.description}</p>
                    <div className="delete"><button className="shadow animate orange" onClick={() => props.onDelete(props.puppies._id)}>Delete</button></div>
                </li>
            </div>
        );
}

