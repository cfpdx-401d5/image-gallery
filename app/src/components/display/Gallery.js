import React from 'react';

export default function GalleryDisplay(props) {
    return (
      <div>
        <h1>Title: {props.resource.title}</h1>
        <img className='gallery' src={props.resource.url} alt='Cute Bunny'/>
        <p>Description: {props.resource.description}</p>
        <button onClick={() => props.onDelete(props.resource._id)}>delete</button>
      </div>
    ); 
};
