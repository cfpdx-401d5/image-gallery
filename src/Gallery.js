import React, { PropTypes } from 'react';

function Gallery(props) {
  let altText = props.title + ' gallery view';

  return (
    <figure>
      <img src={props.url} alt={altText} />
      <figcaption>
        {props.title} <br/>
        {props.description}
      </figcaption>
    </figure>
  );
}
Gallery.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default Gallery;
