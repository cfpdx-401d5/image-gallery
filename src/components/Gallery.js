import React from 'react'

function Gallery(props) {
    return (
        <img alt={props.image.title} src={props.image.url} />
    );
}

export default function GalleryView(props) {
    return (
        <div className={'mainContent'}>
            {props.images.map(image => (
                <li key={image.id}>
                    <Gallery image={image} />
                </li>
            ))}
        </div>
    );
}