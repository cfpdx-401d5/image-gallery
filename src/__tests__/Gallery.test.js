/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Gallery from '../Gallery';

describe('Gallery component', function () {
    it('renders the Gallery component', () => {
        const rendered = shallow(<Gallery url='_url' title='_title' description='_description'/>);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});
