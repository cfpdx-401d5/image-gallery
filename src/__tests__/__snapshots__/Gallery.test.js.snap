exports[`Gallery component renders the Gallery component 1`] = `
<figure>
  <img
    alt="_title gallery view"
    src="_url" />
  <figcaption>
    _title
     
    <br />
    _description
  </figcaption>
</figure>
`;
